src/schemaGenerator.ts

```diff
@@ -114,6 +113,12 @@ function isDocumentNode(typeDefinitions: ITypeDefinitions): typeDefinitions is D
   return (<DocumentNode>typeDefinitions).kind !== undefined;
 }
 
+function uniq(array: Array<any>): Array<any> {
+  return array.reduce((accumulator, currentValue) => {
+    return (accumulator.indexOf(currentValue) === -1) ? [...accumulator, currentValue] : accumulator;
```

Comment:
This is a quadratic implementation of deduplication for strings - might be better to use an object as the accumulator (`accumulator[currentValue] = 1` or similar).

In practice probably not a huge deal unless the schema is split into very many parts.